### DISCUSSION TITLE
Post Count

### DISCUSSION THREAD

**Discussion by AlexHedley** [statiqdev/Discussions#230] [Q&A] [answered]
If I wanted to create a custom page to show some stats is there a way to get the total post count?

I must be missing something obvious:
```cs
@(Outputs.FromPipeline(nameof(Content)).Flatten().FilterSources($"posts/*")).Count()
```
> Statiq.Common.FilteredDocumentList`1[Statiq.Common.IDocument].Count()

_Code_

- [FilteredDocumentList{TDocument}.cs](https://github.com/statiqdev/Statiq.Framework/blob/main/src/core/Statiq.Common/Documents/FilteredDocumentList%7BTDocument%7D.cs)
`src/core/Statiq.Common/Documents/FilteredDocumentList{TDocument}.cs`

---

For each Year I have, which works great.

```cs
@* Years Count *@
@foreach(IGrouping<int, IDocument> group in Outputs
                                            .FromPipeline(nameof(Content))
                                            .Flatten()
                                            .FilterSources($"posts/*")
                                            .GroupBy(x => x.GetDateTime(WebKeys.Published).Year)
                                            .OrderByDescending(x => x.Key))
{
    <h1 class="bg-dark text-light p-2">@group.Key <span class="postCount">@group.Count()</span></h1>
}
```

**Comment by girlpunk** [comment, marked as answer]
Your initial code looks correct, bar an incorrect bracket placement. We can rewrite it as:
```csharp
@(
    Outputs.FromPipeline(
        nameof(Content)
    )
    .Flatten()
    .FilterSources($"posts/*")
)
.Count()
```
Moving that penultimate closing bracket to after the count call should resolve the issue, like so:
```csharp
@(Outputs.FromPipeline(nameof(Content)).Flatten().FilterSources($"posts/*").Count())
```

**Reply by AlexHedley** [reply]
D’oh, thanks for spotting that!
